flocker/node/testtools.py

```diff
@@ -248,7 +248,7 @@ def ideployer_tests_factory(fixture):
 
     :return: ``TestCase`` subclass that will test the given fixture.
     """
-    class IDeployerTests(TestCase):
+    class IDeployerTests(AsyncTestCase):
```

Comment:
As discussed on Slack, `test.reactor` isn't anything special, it's just a guaranteed way of the tests getting the running reactor, so there's no _need_, but it's probably good practice. Relying on globals sucks.

I went to fix this but discovered:
* `test_blockdevice` tests use `NonReactor`
* `test_testtools` tests don't use a reactor at all
* `test_deploy` tests mostly duplicate `test_testtools` tests

Thus, I removed the duplication, and updated the test subclasses to use `SynchronousDeferredRunTest`, which handles tests that return Deferreds but don't use the reactor. 

It's not a thing we should commonly use—tests that want that should use `successResultOf` and friends—but it's a helpful check here to see whether these tests have any hidden dependencies on the reactor.